sycl/test-e2e/KernelCompiler/sycl_and_cache.cpp

```diff
@@ -1,123 +0,0 @@
-//==- sycl_and_cache.cpp - cache works with kernel_compiler sycl  ----------==//
```

Comment:
This test should be kept, no?

This test actually tests that the persistent cache writing and retrieval is working for the kernel_compiler.  The other sycl_cache.cpp  tests that the cache eviction is working.  
I believe this should work with the SYCL JIT variant now that it's being promoted.  

Or am I overlooking something?

Reply:
Good point, it makes sense to bring the test back in a slightly modified form. The story for testing the cache features then is:
- `sycl_cache.cpp` (fka `kernel_compiler_sycl_jit_cache.cpp`): Caching of the device code IR (BLAKE3 hash of preprocessed source, read/write bitcode files after frontend)
- `sycl_cache_pm.cpp` (fka `sycl_and_cache.cpp`): Changed to test that the program manager's persistent caching of native programs also works for `source_language::sycl`
- `opencl_cache_eviction.cpp`: Unchanged, tests the persistent cache for other source languages (and also the eviction)